tests/Unit/AcceleratorViewCopy/copy_coherency.cpp

```diff
@@ -48,19 +49,53 @@ int main()
     int * B  = hc::am_alloc(sizeElements, acc, 0);
     int * Bh = hc::am_alloc(sizeElements, acc, amHostPinned);
 
-    const int binit = 13;
-    //memsetIntKernel(av, A, -1, numElements);
-    memsetIntKernel(av, Bh, -3, numElements);
-    av.wait();
 
-    memsetIntKernel(av, B, -42, numElements);
-    memsetIntKernel(av, B, binit, numElements);
+    if (1) {
+        printf ("test: running same-stream copy coherency test\n");
+        // Reset values:
+        const int bexpected = 13;
+        memset(Bh, 0xAF, numElements); // dummy values to ensure we can tell if copy occurs
+        av0.wait();
+        printf ("test:   setup complete\n");
 
-    // check kernel followed by D2H copy:
-    av.copy_async(B, Bh, sizeElements); 
+        // Set B to -42, followed immediately by setting to expected value:
+        memsetIntKernel(av0, B, -42, numElements);
+        memsetIntKernel(av0, B, bexpected, numElements);
 
-    av.wait();
-    check(Bh, numElements, binit);
+        // Async copy back to host with no intervening code - 
+        // This should fail unless we properly issue a release fence after the memset:
+        av0.copy_async(B, Bh, sizeElements); 
+
```

Comment:
If memory is pinned and read on host as you describe, then we would need a system-scope release to detect this.  The HCC runtime will insert a system-scope release on for accelerator_view::wait() or completion_future::wait().   This is tracked by _needs_sys_release (set if a kernel is dispatched to a qeueue which may dirty the L2) and the necessary release fence is determined by detectStreamDeps.  I will expand the test to include this case.